components/nimbus/src/stateful/nimbus_client.rs

```diff
@@ -704,6 +719,24 @@ impl NimbusClient {
         }
         Ok(())
     }
+
+    #[cfg(test)]
+    pub fn get_metrics_handler(&self) -> &&TestMetrics {
+        let metrics = &**self.metrics_handler;
+        unsafe { std::mem::transmute::<&&dyn MetricsHandler, &&TestMetrics>(&metrics) }
```

Comment:
Please add a `// SAFETY: ...` comment explaining why this is safe.

See https://std-dev-guide.rust-lang.org/policy/safety-comments.html